### PR TITLE
Fix precision to account for small wei increase.

### PR DESCRIPTION
**Pull request by 2-am-zzz**
May be related to the 'off by one wei' issue, but as I continue to work, highly unlikely. Still found this and put in a fix.

### CODE REVIEW

**Comment by danfinlay** [comment]
It would help to have a test that failed before this fix commit, so a reviewer could plainly see what expectation was not being met, but now is.

Would you mind adding unit tests to this, since it's a pretty important piece of code? Our users' ability to enter numbers reliably can't be overstated enough.

**Comment by Zanibas** [comment]
Done--is this sufficient?

File: test/unit/components/bn-as-decimal-input-test.js [hunk 1]
```diff
@@ -48,4 +48,40 @@ describe('BnInput', function () {
...
+
+    const value = new BN(valueStr, 10)
+
+    const inputStr = '1000000000.000000001'
+
+    let targetStr = '1000000000000000001'
+
+    const target = new BN(targetStr, 10)
+
+    const precision = 9 // ether precision
```
Comment (danfinlay) [inline test/unit/components/bn-as-decimal-input-test.js]: Slight misnomer for the test, ether precision is 18, but the test should still work at other scales.

**Review by danfinlay — APPROVED** [review]
That test really helps explain what this PR does, thanks. Does this fix the bug where gas price would sometimes be incremented by 1 wei?

**Comment by danfinlay** [comment]
Probably want to bump changelog, too btw.

**Comment by Zanibas** [comment]
Hold on, not sure why this test passes on master, checking what's actually going on.

**Comment by Zanibas** [comment]
So the state of the component was updating properly, but since it's not connected to a parent component's state that would allow it to re-render, we weren't able to test whether or not the final input after all state changes was unchanged. This test identifies that.

**Pull request merged**